### DISCUSSION TITLE
V5 to v6 migration is not going smoothly

### DISCUSSION THREAD

**Discussion by ChaosYoda** [GitTools/GitVersion#4352] [Q&A] [answered]
Hi all, 

I am doing a v5 to v6 migration and things are not going well. I am using Azure yaml pipelines and trying to build my C# nuget packages, with the removal of NuGetVersionV2, my nuget version is a mess. Previously it would also create a 0 padded commit version, so we could have something like 9.0.8-<branch>0001 , where the 0001 is the number of commits, but now I just have 9.0.8-<branch>-git-version.1+13, now nuget does not understand or accept the +13, so what is the correct way to deal with this now?

**Comment by HHobeck** [comment, marked as answer]
The following example is public available (in the documentation of gitversion) and can be found under [Reference -> Version Variables](https://gitversion.net/docs/reference/variables):

```json
{
    "Major": 3,
    "Minor": 22,
    "Patch": 11,
    "PreReleaseTag": "beta.99",
    "PreReleaseTagWithDash": "-beta.99",
    "PreReleaseLabel": "beta",
    "PreReleaseLabelWithDash": "-beta",
    "PreReleaseNumber": 99,
    "WeightedPreReleaseNumber": 1099,
    "BuildMetaData": 88,
    "FullBuildMetaData": "99.Branch.release/3.22.11.Sha.28c853159a46b5a87e6cc9c4f6e940c59d6bc68a",
    "MajorMinorPatch": "3.22.11",
    "SemVer": "3.22.11-beta.99",
    "AssemblySemVer": "3.22.11.0",
    "AssemblySemFileVer": "3.22.11.0",
    "InformationalVersion": "3.22.11-beta.99+88.Branch.release/3.022.011.Sha.28c853159a46b5a87e6cc9c4f6e940c59d6bc68a",
    "FullSemVer": "3.22.11-beta.99+88",
    "BranchName": "release/3.022.011",
    "EscapedBranchName": "release-3.022.011",
    "Sha": "28c853159a46b5a87e6cc9c4f6e940c59d6bc68a",
    "ShortSha": "28c8531",
    "VersionSourceSha": "28c853159a46b5a87e6cc9c4f6e940c59d6bc68a",
    "CommitsSinceVersionSource": 7,
    "CommitDate": "2021-12-31",
    "UncommittedChanges": 0
}
```

Maybe you need to use the `SemVer` variable instead of the `FullSemVer` variable. Please also keep in mind that the output is dependent on the used deployment mode:
- [Manual Deployment](https://gitversion.net/docs/reference/modes/manual-deployment)
- [Continuous Delivery](https://gitversion.net/docs/reference/modes/continuous-delivery)
- [Continuous Deployment](https://gitversion.net/docs/reference/modes/continuous-deployment)

Happy branching!

**Comment by ChaosYoda** [comment]
The problem is more with that the NuGetPreReleaseTagV2 and NuGetVersionV2 have been removed. The FullSemVer comes out as something like ` "FullSemVer": "9.0.8-141871-git-version.1+15",` , where nuget versioning will ignore the +15, previously in v5, there was the CommitsSinceVersionSourcePadded available which would work with nuget. So I would get a pre-release version like 9.0.7-141167-dex-param0008, but now I get a pre-release version like 9.0.8-141871-git-version.1 , so any further commits to the pre-release (or feature) branch will not increment the version with nuget. For normal release builds, this is not a problem. 
Also the Semver now looks different as well. V5 it was ` "SemVer": "9.0.7-141167-dex-param-method.8",` 
V6 is it now "SemVer": "9.0.8-141871-git-version.1"` (where this is already 15 commits later, so should it not be .15 instead of .1?)  completely missing the CommitsSinceVersionSource information. Or the CommitsSinceVersionSource as in V5.

**Reply by HHobeck** [reply]
I think you need to understand the different between [Manual Deployment](https://gitversion.net/docs/reference/modes/manual-deployment) and [Continuous Delivery](https://gitversion.net/docs/reference/modes/continuous-delivery) deployment mode. If you want to increment the pre-release number in manual deployment mode then you need to create a pre-release tag. If you publishing a nuget package on every build then `Continues Delivery` mode is a better option.

The manual deployment mode documentation states:

> GitVersion will build the same semantic version until that version is deployed.
